### PR TITLE
Solution, that was hard

### PR DESCRIPTION
**Pull request by Lentyaeev**
[DEMO LINK](https://lentyaeev.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,223 +1,244 @@
...
-
-            <span data-cy="TodoTitle" className="todo__title">React</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
+          <TransitionGroup>
```
Comment (roman-romanenko) [inline src/App.tsx:148]: Nice animation 🤩

**Review by roman-romanenko — APPROVED** [review]
Goood job 👍🏻

File: src/App.tsx [hunk 2]
```diff
@@ -1,223 +1,244 @@
...
+  const clearCompleted = () => {
+    completedTodos.forEach(todo => {
+      onDelete(todo.id);
+    });
+  };
+
+  const onSubmit = () => {
+    if (todoTitle.length === 0) {
+      return;
+    }
```
Comment (VETALEK) [inline src/App.tsx:101]:
```suggestion
    if (todoTitle.trim.length === 0) {
      return;
    }
```
image

File: src/App.tsx [hunk 3]
```diff
@@ -1,223 +1,244 @@
...
+    toToggle.forEach(todo => {
+      patchTodo({ ...todo, completed: !todo.completed });
+    });
+  };
+
+  const clearCompleted = () => {
+    completedTodos.forEach(todo => {
+      onDelete(todo.id);
+    });
+  };
```
Comment (VETALEK) [inline src/App.tsx:96]: let's implement it with `Promise.all`
image

and don't forget about `useCallback`

**Review by VETALEK — CHANGES_REQUESTED** [review]
GJ 👍 
commented

**Review by VETALEK — APPROVED** [review]
GJ 🔥

**Pull request closed without merging**